lib/searchkick/query.rb

```diff
@@ -191,7 +191,7 @@ def handle_response(response)
     end
 
     def retry_misspellings?(response)
-      @misspellings_below && Results.new(searchkick_klass, response).total_count < @misspellings_below
+      @misspellings_below && response["error"].nil? && Results.new(searchkick_klass, response).total_count < @misspellings_below
```

Comment:
It might make more sense to check `response["hits"].is_a?(Hash)`, or something like that, rather than `response["error"].nil?`. I think that which to choose is mostly a matter of taste/philosophy, and I defer to yours, of course, @ankane (or maybe there's some other approach altogether that you think would be better to avoid the exception that we were able to surface).